include/irr/asset/filters/kernels/CChannelIndependentImageFilterKernel.h

```diff
@@ -80,25 +87,43 @@ class CChannelIndependentImageFilterKernel<KernelR,void,void,void> : public CFlo
 		}
 
 		_IRR_STATIC_INLINE_CONSTEXPR bool has_derivative = KernelR::has_derivative;
-		inline std::enable_if_t<has_derivative,float> d_weight(float x, int32_t channel) const
+		inline float d_weight(float x, int32_t channel) const
```

Comment:
any way to disable this function at compile time if `!has_derivative` @Crisspl ?

I'm not an `enable_if` expert unfortunately.